include/tvm/data_layout.h

```diff
@@ -210,6 +210,78 @@ class Layout : public NodeRef {
     return ct;
   }
 
+  /*!
+   * \brief Returns a new layout where the dims have been expanded to match the primal dimensions.
+   * \param dst_layout The dst layout to which current layout has to be expanded.
+   * \return The expanded Layout.
+   */
+  inline Layout ExpandLayout(const Layout& dst_layout) {
+    Layout new_src_layout;
+    // 1) Find the axis which are missing in the current layout. Make them the prefix.
+    std::string new_src_layout_str = "";
+    for (auto dst_axis : dst_layout->axes) {
+      if (LayoutAxis::Get(dst_axis).IsPrimal()) {
+        if (this->IndexOf(LayoutAxis::Get(dst_axis)) == -1) {
+          new_src_layout_str += dst_axis->var->name_hint;
+        }
+      }
+    }
+    // 2) Now, add the primal axis of the current layout.
+    for (auto src_axis : operator->()->axes) {
+      if (LayoutAxis::Get(src_axis).IsPrimal()) {
+        new_src_layout_str += src_axis->var->name_hint;
+      }
+    }
+    new_src_layout = Layout(new_src_layout_str);
+    return new_src_layout;
+  }
+
+  /*!
+   * \brief Returns a new layout where the subordinate factors are adjusted based on the tensor
+   *        shape.
+   * \param old_layout The old layout before any transformation.
+   * \param old_shape The shape of the original tensor.
+   * \return The adjusted Layout.
+   */
+  inline Layout AdjustSubordinateFactors(const Layout& old_layout,
+                                         const Array<tvm::Expr>& old_shape) const {
+    // For each subordinate axis
+    //   1) Find the corresponding dual axis.
+    //   2) Find the Index of this dual axis in old_layout.
+    //   3) Find the shape of the that axis in old_shape.
+    //   4) a) Adjust factor to 1, if that shape is 1. b) Else retain the factor.
+    std::string new_small_layout;
+    for (auto axis : operator->()->axes) {
+      if (!LayoutAxis::Get(axis).IsPrimal()) {
+        // 1) Find the corresponding dual axis
+        auto dual_axis = std::toupper(LayoutAxis::Get(axis).name()[0]);
```

Comment:
LayoutAxis::Get(axis).ToPrimal() or ToDual()

Reply:
The function is private. Can't do that.